### DISCUSSION TITLE
Best practice for data fetching for Nuxt 3

### DISCUSSION THREAD

**Discussion by Nikolos56651** [nuxt/nuxt#16608] [Questions] [answered]
Hello,

I have to take data from a REST api also made by me. I did various research but I did not find any good practice regarding data fetching.

Currently I was thinking of doing a fetch directly on the REST API from the via page and use nginx (for example) to redirect to the rest api but I saw that it was possible to make a file in the /server folder that could fetch it on the REST API.
What is the best practice in terms of security and performance ?
Is there a big difference for the customer (in case of traffic inspection,...) ?
And does this apply in all cases (select data, update data and different case) ?

**Comment by procrates** [comment, marked as answer]
Here you go: [https://v3.nuxtjs.org/getting-started/data-fetching](https://v3.nuxtjs.org/getting-started/data-fetching)

If you need to fetch data with a private key: [https://v3.nuxtjs.org/guide/directory-structure/server](https://v3.nuxtjs.org/guide/directory-structure/server)

**Reply by Nikolos56651** [reply]
Thanks you for your answer, i already read the document but in the document, all data fetch request the api in the /server. 
The only one who don't make that is to call an external api in another website.  In my case, i need to call an internal api server. 
Do you suggest to call the api in /server and this api call the another rest api, or directly call the rest api in the first fetch ?

**Reply by TheAlexLichter** [reply]
If you don't need a "hop" in between (e.g. for caching - see also "BFF") then call your API directly ☺️

**Reply by Nikolos56651** [reply]
Thanks, I'll go with a direct call from the REST api
